src/main/java/io/odpf/firehose/sink/clickhouse/QueryTemplate.java

```diff
@@ -0,0 +1,118 @@
+        this.scopes = new HashMap<>();
+        this.kafkaRecordParserMode = clickhouseSinkConfig.getKafkaRecordParserMode();
+        initialize(clickhouseSinkConfig);
+        buildQuery();
+    }
+
+    private void buildQuery() {
+        String query = INSERT_QUERY;
+        template = Mustache.compiler().withEscaper(Escapers.simple()).compile(query);
+    }
+
+    private void initialize(ClickhouseSinkConfig clickhouseSinkConfig) {
+        scopes.put("table", clickhouseSinkConfig.getClickhouseTableName());
+
+        Properties messageProtoToDBColumnsMapping = clickhouseSinkConfig.getInputSchemaProtoToColumnMapping();
+        insertColumns = getInsertColumns(messageProtoToDBColumnsMapping);
+        scopes.put("insertColumns", String.join(",", insertColumns));
+    }
+
+    private List<String> getInsertColumns(Properties messageProtoToDBColumnsMapping) {
+        List<String> columns = new ArrayList<>();
+        Enumeration<?> propertyNames = messageProtoToDBColumnsMapping.propertyNames();
+        while (propertyNames.hasMoreElements()) {
+            Object tableColumn = messageProtoToDBColumnsMapping.get(propertyNames.nextElement());
+            if (tableColumn instanceof String) {
+                columns.add((String) tableColumn);
+            } else if (tableColumn instanceof Properties) {
+                columns.addAll(getInsertColumns((Properties) tableColumn));
+            }
+        }
+        return columns;
+    }
+
+    public String toQueryStringForSingleMessage(Message message) {
+
+        byte[] value;
+
+        if ("message".equals(kafkaRecordParserMode)) {
+            value = message.getLogMessage();
+        } else {
+            value = message.getLogKey();
+        }
+
+        Map<String, Object> columnToValue = protoToFieldMapper.getFields(value);
+
+        String insertValues = stringifyColumnValues(columnToValue, insertColumns);
+
+        scopes.put("insertValues", insertValues);
+
+        return template.execute(scopes);
+    }
+
+    private String stringifyColumnValues(Map<String, Object> columnToValue, List<String> columns) {
+        List<String> columnValues = columns.stream()
+                .map(c -> columnToValue.get(c).toString().replace("'", "''"))
+                .map(c -> "\'" + c + "\'")
+                .collect(Collectors.toList());
+        return String.join(", ", columnValues);
+    }
+
+    public String toQueryStringForMultipleMessages(List<Message> messages) {
+        byte[] value;
+        String insertValues = "";
+
+        long count = 0; //counter to maintain number of rows in the query.
+        for (Message message : messages) {
+            if ("message".equals(kafkaRecordParserMode)) {
+                value = message.getLogMessage();
+            } else {
+                value = message.getLogKey();
+            }
+            count++;
+            Map<String, Object> columnToValue = protoToFieldMapper.getFields(value);
```

Comment:
isn't count just a for loop variable? 
why not use `for (int index = 0; index < messages.getSize(); index++)`